mod_reforged/config/indirect_numeral_adjectives.nut

```diff
@@ -0,0 +1,202 @@
+::Reforged.Numerals <- {
+	// Config
+	ExactPartySize = false,
+	ExactDialogSize = false,
+	PartyRange = false,
+	DialogRange = true
+
+	// Data
+	IndefiniteNumerals = [],
+
+	function register( _name, _startsAt, _suffix = "" )
+	{
+		::Reforged.Numerals.IndefiniteNumerals.push({
+			Name = _name,
+			StartsAt = _startsAt,
+			Suffix = _suffix,
+			Range = ""
+		});
+	}
+
+	function setExactPartySize( _bool )
+	{
+		this.ExactPartySize = _bool;
+		this.updateAllParties();
+	}
+
+	function setPartyRange( _bool )
+	{
+		this.PartyRange = _bool;
+		this.updateAllParties();
+	}
+
+
+	// This always needs to be called after all new numerals of your mod are registered.
+	// This will sort all Numerals and generate their ranges automatically
+	function calculateRanges()
+	{
+		this.IndefiniteNumerals.sort(@(a, b) a.StartsAt <=> b.StartsAt);
+
+		local minAmount = 1;
+		local maxAmount = 1;
+
+		for (local i = 1; i < this.IndefiniteNumerals.len(); i++)
+		{
+			maxAmount = (this.IndefiniteNumerals[i].StartsAt - 1);
+			if (minAmount != maxAmount || minAmount != 1)	// We hard-code that the (1) for One would never displayed
+			{
+				this.IndefiniteNumerals[i - 1].Range = "" + minAmount;
+				if (minAmount != maxAmount) this.IndefiniteNumerals[i - 1].Range += "-" + maxAmount;
+			}
+			minAmount = maxAmount + 1;
+		}
+		this.IndefiniteNumerals.top().Range = minAmount + "+";
+	}
+
+	function getNumeralString( _amount, _isWorldParty )
+	{
+		if (this.IndefiniteNumerals.len() == 0) return _amount;
+		if (this.ExactPartySize && _isWorldParty) return _amount;
+		if (this.ExactDialogSize && !_isWorldParty) return _amount;
+
+		local numeralEntry = this.getNumeralEntryByAmount(_amount);
+
+		local numeralString = numeralEntry.Name;
+		if (!_isWorldParty) numeralString += numeralEntry.Suffix;
+		if (this.PartyRange && _isWorldParty || this.DialogRange && !_isWorldParty)
+		{
+			if (numeralEntry.Range != "") numeralString += " (" + numeralEntry.Range + ")";
+		}
+		return numeralString;
+	}
+
+	// Can be used by other mods to adjust individuell entries
+	function getNumeralEntryByName( _name )
+	{
+		foreach (numeral in this.IndefiniteNumerals)
+		{
+			if (numeral.Name == _namet) return numeral;
+		}
+		return null;
```

Comment:
`return null` is the default return of every function so this line is redundant.